### ISSUE TITLE
whether one tensor  can be distributed to different gpu cards?

### ISSUE THREAD

**Issue by lw3259111** [tensorflow/tensorflow#19447]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **TensorFlow installed from (source or binary)**:
- **TensorFlow version (use command below)**:
- **Python version**: 
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem

I have six gpu cards ,  i have a problem that my data have 90000 attribute, whether i can be distributed one `weight`  tensor to different cards?  where i can found a example?

**Comment by xysmlx** [comment]
No. In tensorflow's abstraction, one operator with its input/output tensors can only be executed on one device. However, you can achieve it by using the following methods:

1. use the data parallelism mode of tensorflow (parameter server or replication), like distributed tensorflow
2. create weight tensor for each card, and synchronize them manually, example: https://github.com/tensorflow/models/blob/master/tutorials/image/cifar10/cifar10_multi_gpu_train.py

**Comment by lw3259111** [comment]
@xysmlx Thanks for your help. I had tried the `cifar10_multi_gpu_train.py` example,  i only use data with batch_size=8,features=20000. but when use one gpu for backpropagation，the gpu was used 15G. so i can't increase features to 90000.

**Comment by xysmlx** [comment]
@lw3259111 Have you confirmed that the tensorflow returns OOM when increases features to 90000? Tensorflow will allocate all of the GPU memory and manage them by itself in default. The GPU memory shown in nvidia-smi is useless.

If tensorflow returns OOM, there are two ways:
1. decrease the size of the batch_size
2. split a batch into multiple samples and run forward propagation for each sample. Then, calculate the loss for each sample and sum them into one loss. Then, do backward propagation manually with this loss.

Have you see whether features are dense or sparse?  If the features are sparse, it will be better to use sparse operator to process them.

**Comment by reedwm** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!
